common/metrics/otel_event_handler.go

```diff
@@ -60,12 +57,11 @@ func NewOtelMetricHandler(l log.Logger, o OpenTelemetryProvider, cfg ClientConfi
 	return &OtelMetricHandler{
 		provider:    o,
 		l:           l,
-		recordFuncs: map[event.Metric]recordFunc{},
 		excludeTags: configExcludeTags(cfg),
 	}
 }
 
-func (m *OtelMetricHandler) Event(ctx context.Context, e *event.Event) context.Context {
```

Comment:
Can you provide some context on why the pointer is not necessary here?

Reply:
nothing is changing on the receiver and does not need to be a pointer.